addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js

```diff
@@ -0,0 +1,96 @@
+// This Script will seek at incoming WebSocket XML formatted messages for suspicious comments.
+
+// * Is based on: org.zaproxy.zap.extension.pscanrulesBeta.InformationDisclosureSuspiciousComments
+// ** And the comment list is copied by
+// *** https://github.com/zaproxy/zap-extensions/blob/master/addOns/pscanrulesBeta/src/main/zapHomeFiles/xml/suspicious-comments.txt
+
+// Author: Manos Kirtas (manolis.kirt@gmail.com)
+
+OPCODE_TEXT = 0x1;
+RISK_INFO 	= 0;
+CONFIDENCE_MEDIUM = 2;
+
+DocumentBuilderFactory = Java.type("javax.xml.parsers.DocumentBuilderFactory");
+StringReader = Java.type("java.io.StringReader");
+InputSource = Java.type("org.xml.sax.InputSource");
+Node = Java.type("org.w3c.dom.Node");
+Comment = Java.type("org.w3c.dom.Comment");
+
+commentPatterns = [ /\bTODO\b/gmi,
+                  /\bFIXME\b/gmi,
+                  /\bBUG\b/gmi,
+                  /\bBUGS\b/gmi,
+                  /\bXXX\b/gmi,
+                  /\bQUERY\b/gmi,
+                  /\bDB\b/gmi,
+                  /\bADMIN\b/gmi,
+                  /\bADMINISTRATOR\b/,
+                  /\bUSER\b/gmi,
+                  /\bUSERNAME\b/gmi,
+                  /\bSELECT\b/gmi,
+                  /\bWHERE\b/gmi,
+                  /\bFROM\b/gmi,
+                  /\bLATER\b/gmi
+                ];
+
+function scan(helper,msg) {
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing){
+        return;
+    }
+
+    var message = String(msg.getReadablePayload()).valueOf();
+    xmlDoc = getParsedDocument(message);
+
+    if(xmlDoc == null){
+        return;
+    }
+
+    var counter = 0;
+    var commentsList = [];
+    getComments(xmlDoc.getDocumentElement(), commentsList);
+
+    commentsList.forEach(function(comment){
+        commentPatterns.forEach(function(pattern){
+            if(pattern.test(comment)){
+                helper.newAlert()
+                    .setRiskConfidence(RISK_INFO, CONFIDENCE_MEDIUM)
+                    .setName("Information Disclosure - Suspicious Comments in XML via WebSocket (script)")
+                    .setDescription("The response appears to contain suspicious comments which may help an attacker.")
+                    .setParam(counter++)
```

Comment:
I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

Reply:
Fixed in 42d6098cb